tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -34,6 +41,352 @@
+   *
+   * <p>This method copies
+   * <i>n</i>&nbsp;=&nbsp;<tt>src.remaining()</tt> elements from the given
+   * buffer into the tensor, starting at the buffer's current position.
+   * The position of the buffer is then incremented by <i>n</i>.
+   *
+   * @param dataType the tensor datatype.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   *
+   * @throws  IllegalArgumentException
+   *          If the tensor datatype or shape is not compatible with the buffer
+   */
+  public static Tensor create(DataType dataType, long[] shape, LongBuffer data) {
+    if(dataType != DataType.INT64) {
+      throw incompatibleBuffer(dataType);
+    }
+    if(data.remaining() != numElements(shape)) {
+      throw incompatibleBuffer(shape);
+    }
+    int elemSize = Long.SIZE / Byte.SIZE;
+    Tensor t = create(dataType, shape, elemSize * data.remaining());
+    try {
+      ByteBuffer dst = t.buffer();
+      dst.asLongBuffer().put(data);
+      return t;
+    }
+    catch(RuntimeException e) {
+      delete(t.nativeHandle);
+      throw e;
+    }
+  }
+
+  /**
+   * Create a Tensor with data from the given buffer.
+   *
+   * <p>Supports all datatypes.  Note that primitive data must be in native byte order.
+   *
+   * <p>This method copies
+   * <i>n</i>&nbsp;=&nbsp;<tt>src.remaining()</tt> elements from the given
+   * buffer into the tensor, starting at the buffer's current position.
+   * The position of the buffer is then incremented by <i>n</i>.
+   *
+   * @param dataType the tensor datatype.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   *
+   * @throws  IllegalArgumentException
+   *          If the tensor datatype or shape is not compatible with the buffer
+   */
+  public static Tensor create(DataType dataType, long[] shape, ByteBuffer data) {
+    switch(dataType) {
+      case FLOAT:
+      case DOUBLE:
+      case INT32:
+      case INT64:
+      case BOOL:
+        int elemSize = elemByteSize(dataType);
+        if(data.remaining() / elemSize != numElements(shape)) {
+          throw incompatibleBuffer(shape);
+        }
+        break;
+      case STRING:
+      default:
+        // not all types are checked
+        break;
+    }
+
+    Tensor t = create(dataType, shape, data.remaining());
+    try {
+      ByteBuffer dst = t.buffer();
+      dst.put(data);
+      return t;
+    }
+    catch(RuntimeException e) {
+      delete(t.nativeHandle);
+      throw e;
+    }
+  }
+
+  private static Tensor create(DataType dataType, long[] shape, long byteSize) {
+    Tensor t = new Tensor();
+    t.dtype = dataType;
+    t.shapeCopy = Arrays.copyOf(shape, shape.length);
+    t.nativeHandle = allocate(t.dtype.c(), t.shapeCopy, byteSize);
+    return t;
+  }
+
+  /**
+   * Gets the size (in bytes) of the tensor data.
+   *
+   * <p>Use this size information when constructing a byte buffer for {@code #writeTo(...)}.
+   */
+  public int byteSize() {
```

Comment:
Perhaps call this `numBytes()`, so we will have `numBytes()` and `numElements()`